httprequest_lego_provider/dns.py

```diff
@@ -0,0 +1,63 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+"""DNS utiilities."""
+
+from pathlib import Path
+from tempfile import TemporaryDirectory
+
+from git import Git, GitCommandError, Repo
+
+from .settings import DNS_REPOSITORY_URL, SSH_IDENTITY_FILE
+
+REPOSITORY_BASE_URL = "@".join(DNS_REPOSITORY_URL.split("@")[:-1])
+REPOSITORY_BRANCH = DNS_REPOSITORY_URL.split("@")[-1]
+RECORD_CONTENT = "; GENERATED BY httpreq - DO NOT EDIT!"
+SSH_EXECUTABLE = f"ssh -i {SSH_IDENTITY_FILE}"
+
+
+class DnsSourceUpdateError(Exception):
+    """Exception for DNS update errors."""
+
+
+def write_dns_record(dns_record: str) -> None:
+    """Write a DNS record following the canonical-is-dns-configs specs if it doesn't exist.
+
+    Args:
+        dns_record: the DNS record to add.
+
+    Raises:
+        DnsSourceUpdateError: if an error while updating the repository occurs.
+    """
+    with TemporaryDirectory() as tmp_dir, Git().custom_environment(GIT_SSH_COMMAND=SSH_EXECUTABLE):
+        try:
+            repo = Repo.clone_from(REPOSITORY_BASE_URL, tmp_dir, branch=REPOSITORY_BRANCH)
+            dns_record_file = Path(f"{repo.working_tree_dir}/{dns_record}")
+            if not bool(dns_record_file.exists):
+                dns_record_file.write_text(RECORD_CONTENT, encoding="utf-8")
+                repo.index.add([dns_record])
+                repo.git.commit("-m", f"Add {dns_record} record")
+                repo.remote(name="origin").push()
```

Comment:
Is there a source/spec that describes what action is taken after the record is written and pushed to the branch? 
And could you please update the PR description to include a spec or some more details on the rationale for this PR? Thanks very much.

Reply:
I've linked the spec. Let me know if there's anything else that might be useful